### DISCUSSION TITLE
Computing Project amount of detail required in explaination

### DISCUSSION THREAD

**Discussion by Takodachi6969** [Part-II-Computational-Physics/.github#88] [General]
I am doing my Project on Lenia, which surprisingly has a very easy implementation since 90% of the code were readily available online, so I have decided to write many checks to compare the different models of Lenia.

However it quickly turned into a spaghetti code, which I am struggling to understand myself... 

Do I need to rewrite them into classes and separate them into packages, or should I explain them as it is used in the Jupiter notebook with more text? And how much explanation should I give to functions I wrote?

**Comment by xingjianhou** [comment]
Somewhat unrelated: I wanted to choose Lenia at first but got worried that it's too easy, so high marks may be a bit hard to get. Any ideas of what makes a good computing project?

**Reply by Takodachi6969** [reply]
I’ve emailed Austen with a mini proposal, he said things like Lenia is straight forward with analysis so it’s more exploratory in nature….. so I was just writing automated code to quantitatively test those different models (multiple kernels and stuff) with all the available animals(their lifetime, end state, and so on), and trying to draw conclusions on what they are doing and their limitations. 

Im still not sure what are they looking for in the project, and I e already spent more than 80 hours on it without much of a breakthrough
